presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -872,24 +872,24 @@ folly::Future<std::unique_ptr<Result>> TaskManager::getResults(
       request->token = token;
       request->maxSize = maxSize;
       prestoTask->resultRequests.insert({destination, std::move(request)});
-      return std::move(future).via(eventBase).onTimeout(
+      return std::move(future).via(executor).onTimeout(
           std::chrono::microseconds(maxWaitMicros), timeoutFn);
     }
   } catch (const velox::VeloxException& e) {
     promiseHolder->promise.setException(e);
-    return std::move(future).via(eventBase);
+    return std::move(future).via(executor);
   } catch (const std::exception& e) {
     promiseHolder->promise.setException(e);
-    return std::move(future).via(eventBase);
+    return std::move(future).via(executor);
   }
 }
 
 folly::Future<std::unique_ptr<protocol::TaskStatus>> TaskManager::getTaskStatus(
     const TaskId& taskId,
     std::optional<protocol::TaskState> currentState,
     std::optional<protocol::Duration> maxWait,
```

Comment:
Shall we pass the executor to TaskManager?

Reply:
Yeah I guess we can do that